core/src/main/java/com/linecorp/armeria/server/DefaultServiceConfigSetters.java

```diff
@@ -228,6 +233,23 @@ public ServiceConfigSetters multipartUploadsLocation(Path multipartUploadsLocati
         return this;
     }
 
+    @Override
+    public ServiceConfigSetters serviceWorkerGroup(EventLoopGroup serviceWorkerGroup,
+                                                   boolean shutdownOnStop) {
+        this.serviceWorkerGroup = requireNonNull(serviceWorkerGroup, "serviceWorkerGroup");
+        if (shutdownOnStop) {
+            shutdownSupports.add(ShutdownSupport.of(serviceWorkerGroup));
+        }
+        return this;
+    }
+
+    @Override
+    public ServiceConfigSetters serviceWorkerGroup(int numThreads) {
+        checkArgument(numThreads >= 0, "numThreads: %s (expected: >= 0)", numThreads);
```

Comment:
`0` is not allowed:
https://github.com/line/armeria/blob/42bd0d719409efe0f63bcca97d00f5cb9b66f8f7/core/src/main/java/com/linecorp/armeria/common/util/EventLoopGroups.java#L88

If you want, you can also just delegate the check to `EventLoopGroups.newEventLoopGroup()`.